public/app/plugins/datasource/azuremonitor/components/LogsQueryBuilder/AggregateItem.tsx

```diff
@@ -0,0 +1,156 @@
+import { range } from 'lodash';
+import React, { useState, useEffect } from 'react';
+
+import { SelectableValue } from '@grafana/data';
+import { InputGroup, AccessoryButton } from '@grafana/plugin-ui';
+import { Select, Label } from '@grafana/ui';
+
+import {
+  BuilderQueryEditorExpressionType,
+  BuilderQueryEditorPropertyType,
+  BuilderQueryEditorReduceExpression,
+} from '../../dataquery.gen';
+
+interface AggregateItemProps {
+  aggregate: Partial<BuilderQueryEditorReduceExpression>;
+  columns: Array<SelectableValue<string>>;
+  onChange: (item: BuilderQueryEditorReduceExpression) => void;
+  onDelete: () => void;
+  templateVariableOptions: SelectableValue<string>;
+}
+
+const AggregateItem: React.FC<AggregateItemProps> = ({
+  aggregate,
+  onChange,
+  onDelete,
+  columns,
+  templateVariableOptions,
+}) => {
+  const [isPercentile, setIsPercentile] = useState(aggregate.reduce?.name.includes('percentile'));
+  const [isCountAggregate, setIsCountAggregate] = useState(aggregate.reduce?.name.includes('count'));
+
+  const [percentileValue, setPercentileValue] = useState(aggregate.parameters?.[0]?.value || '');
+  const [columnValue, setColumnValue] = useState(aggregate.property?.name || '');
+
+  useEffect(() => {
+    setIsPercentile(aggregate.reduce?.name.includes('percentile'));
+    setIsCountAggregate(aggregate.reduce?.name.includes('count'));
+  }, [aggregate.reduce?.name]);
+
+  const safeTemplateVariables: Array<SelectableValue<string>> =
+    templateVariableOptions && templateVariableOptions.value
+      ? Array.isArray(templateVariableOptions)
+        ? templateVariableOptions
+        : [templateVariableOptions]
+      : [];
+
+  const selectableOptions = columns.concat(safeTemplateVariables);
+
+  const updateAggregate = (percentile?: string, column?: string) => {
+    onChange({
+      ...aggregate,
+      reduce: {
+        name: 'percentile',
```

Comment:
Should this always be set to percentile?

Reply:
if `isPercentile` is true, yes